server/src/main/java/io/unitycatalog/server/persist/ExternalLocationRepository.java

```diff
@@ -0,0 +1,195 @@
+package io.unitycatalog.server.persist;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.model.CreateExternalLocation;
+import io.unitycatalog.server.model.ExternalLocationInfo;
+import io.unitycatalog.server.model.ListExternalLocationsResponse;
+import io.unitycatalog.server.model.UpdateExternalLocation;
+import io.unitycatalog.server.persist.dao.ExternalLocationDAO;
+import io.unitycatalog.server.persist.dao.StorageCredentialDAO;
+import io.unitycatalog.server.persist.utils.PagedListingHelper;
+import io.unitycatalog.server.utils.IdentityUtils;
+import io.unitycatalog.server.utils.ValidationUtils;
+import java.util.*;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class ExternalLocationRepository {
+  private final Repositories repositories;
+  private final SessionFactory sessionFactory;
+  private static final PagedListingHelper<ExternalLocationDAO> LISTING_HELPER =
+      new PagedListingHelper<>(ExternalLocationDAO.class);
+
+  public ExternalLocationRepository(Repositories repositories, SessionFactory sessionFactory) {
+    this.repositories = repositories;
+    this.sessionFactory = sessionFactory;
+  }
+
+  public ExternalLocationDAO addExternalLocation(CreateExternalLocation createExternalLocation) {
+    ValidationUtils.validateSqlObjectName(createExternalLocation.getName());
+    String callerId = IdentityUtils.findPrincipalEmailAddress();
+
+    try (Session session = sessionFactory.openSession()) {
+      StorageCredentialDAO storageCredentialDAO =
+          repositories
+              .getStorageCredentialRepository()
+              .getStorageCredentialDAO(session, createExternalLocation.getCredentialName());
+      UUID externalLocationId = UUID.randomUUID();
+      ExternalLocationDAO externalLocationDAO =
+          ExternalLocationDAO.builder()
+              .id(externalLocationId)
```

Comment:
does DB UC create unique ids in this way too? that is the structure of id string returned by DBUC same as that will be returned here?